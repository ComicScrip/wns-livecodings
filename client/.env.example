REACT_APP_GRAPHQL_API_URL=http://localhost:4000