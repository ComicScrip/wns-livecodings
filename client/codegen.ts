import type { CodegenConfig } from "@graphql-codegen/cli";

const config: CodegenConfig = {
  overwrite: true,
  schema: "http://localhost:4000",
  documents: "src/gql/*.{gql,graphql}",
  generates: {
    "src/gql/generated.ts": {
      preset: "client",
      plugins: ["typescript-react-apollo"],
    },
  },
};

export default config;
