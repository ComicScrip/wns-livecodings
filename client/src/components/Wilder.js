import blank_profile from '../assets/avatar.png';
import Skill from './Skill';

const Wilder = ({ name, city, skills }) => {
  return (
    <article className='card'>
      <img src={blank_profile} alt='Jane Doe Profile' />
      <h3>{name[0].toUpperCase() + name.split('').splice(1).join('')}</h3>
      <h4>{city}</h4>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
        commodo consequat.
      </p>
      <h4>Wild Skills</h4>
      <ul className='skills'>
        {skills.map((skill, index) => (
          <Skill key={index} title={skill.title} votes={skill.votes} />
        ))}
      </ul>
    </article>
  );
};

export default Wilder;
