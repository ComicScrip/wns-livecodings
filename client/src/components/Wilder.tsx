import React from "react";
import blank_profile from "../assets/avatar.png";
import Skill from "./Skill";
import { Link } from "react-router-dom";
import {
  useDeleteWilderMutation,
  useGetProfileQuery,
  Wilder as WilderType,
  WildersDocument,
} from "../gql/generated/schema";

interface WilderProps {
  wilder: WilderType;
}

const Wilder = ({
  wilder: { id, name, skills = [], avatarUrl },
}: WilderProps) => {
  const [deleteWilder] = useDeleteWilderMutation({
    refetchQueries: [{ query: WildersDocument }],
  });

  const handleDelete = async () => {
    if (window.confirm("are you sure ?"))
      try {
        await deleteWilder({ variables: { deleteWilderId: id } });
      } catch (err) {
        console.error(err);
      }
  };

  const { data: currentUser } = useGetProfileQuery();
  const canDeleteWilder = currentUser?.profile.role === "admin";

  return (
    <>
      <div className="flex bg-white p-4 rounded-2xl mb-4 shadow-md">
        <Link to={`/wilders/${id}`}>
          <img
            src={avatarUrl || blank_profile}
            alt={name}
            className="h-16 w-16 rounded-full mr-6"
          />
        </Link>

        <div className="flex justify-between w-full  min-w-[200px]">
          <div className="flex flex-col">
            <Link to={`/wilders/${id}`}>
              <h3 className="font-semibold">
                {name[0].toUpperCase() + name.split("").splice(1).join("")}
              </h3>
            </Link>

            <ul className="flex flex-wrap">
              {skills
                .slice()
                .sort((a, b) => b.votes - a.votes)
                .map((skill, index) => (
                  <Skill
                    key={index}
                    title={skill.name}
                    votes={skill.votes}
                    wilderId={id}
                    skillId={skill.id}
                  />
                ))}
            </ul>
          </div>
          <div className="flex flex-col min-w-[40px]">
            <Link to={`/wilders/${id}/edit`}>
              <button className="mb-2 w-full">✏️</button>
            </Link>
            {canDeleteWilder && <button onClick={handleDelete}>x</button>}
          </div>
        </div>
      </div>
    </>
  );
};

export default Wilder;
