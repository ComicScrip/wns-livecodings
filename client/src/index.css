@tailwind base;
@tailwind components;
@tailwind utilities;

html,
body, #root {
  height: 100%;
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: flex;
  flex-direction: column;
  @apply bg-cream
}

main {
  flex-grow: 1;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}

body {
  font-family: sans-serif;
  color: #222;
  margin: 0;
}

h1,
h2,
h3 {
  margin: 0 0 0.35em;
}

h1 {
  font-size: 40px;
}

h2 {
  font-size: 28px;
}

h3 {
  font-size: 20px;
}

h4 {
  font-size: 15px;
}

p {
  margin: 0 0 1.15em;
}

.container {
  @apply pr-4 pl-4 m-auto max-w-[800px]
}

header, footer {
  @apply bg-pink text-white
}

button, button[type="submit"] {
  @apply cursor-pointer bg-pink font-semibold border border-pink hover:bg-pinkIntense text-white p-2 pr-3 pl-3 rounded-2xl outline-none hover:scale-[102%] max-h-11
}

input, textarea {
  @apply p-2 outline-none border border-gray-200 rounded-2xl
}

label {
  cursor: pointer
}