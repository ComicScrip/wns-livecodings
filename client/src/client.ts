import { ApolloClient, InMemoryCache } from "@apollo/client";

export default new ApolloClient({
  uri: "http://localhost:4000",
  cache: new InMemoryCache(),
  defaultOptions: {
    query: {
      fetchPolicy: "cache-first",
    },
  },
});
