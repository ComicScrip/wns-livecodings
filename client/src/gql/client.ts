import { ApolloClient, createHttpLink, InMemoryCache } from "@apollo/client";

//https://www.apollographql.com/docs/react/networking/authentication/#cookie
export default new ApolloClient({
  cache: new InMemoryCache(),
  defaultOptions: {
    query: {
      fetchPolicy: "cache-first",
    },
  },
  link: createHttpLink({
    uri: process.env.REACT_APP_GRAPHQL_API_URL,
    credentials: "include",
  }),
});
