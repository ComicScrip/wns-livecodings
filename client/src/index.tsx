import React from "react";
import ReactDOM from "react-dom/client";
import "./index.css";
import App from "./App";
import reportWebVitals from "./reportWebVitals";
import { BrowserRouter } from "react-router-dom";
import { MuiThemeProvider, createTheme } from "@material-ui/core/styles";
import { ApolloProvider } from "@apollo/client";
import client from "./gql/client";

const theme = createTheme({
  palette: {
    primary: {
      main: "#F76C6C",
    },
  },
});

const root = ReactDOM.createRoot(
  document.getElementById("root") as HTMLElement
);
root.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <BrowserRouter>
        <MuiThemeProvider theme={theme}>
          <App />
        </MuiThemeProvider>
      </BrowserRouter>
    </ApolloProvider>
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
