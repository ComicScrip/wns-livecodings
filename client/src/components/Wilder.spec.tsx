import React from "react";
import { render, screen } from "@testing-library/react";
import Wilder from "./Wilder";
import { MockedProvider } from "@apollo/client/testing";
import "@testing-library/jest-dom/extend-expect";
import { BrowserRouter } from "react-router-dom";

test("renders Wilder with name and skills", () => {
  const view = render(
    <MockedProvider mocks={[]} addTypename={false}>
      <Wilder
        wilder={{
          id: 1,
          name: "Dave",
          skills: [
            { id: 1, name: "PHP", votes: 3 },
            { id: 1, name: "JS", votes: 7 },
          ],
        }}
      />
    </MockedProvider>,
    { wrapper: BrowserRouter }
  );
  expect(screen.getByText(/Dave/)).toBeInTheDocument();
  expect(screen.getByText(/PHP/)).toBeInTheDocument();
  expect(screen.getByText(/JS/)).toBeInTheDocument();
  expect(screen.getByText(/3/)).toBeInTheDocument();
  expect(screen.getByText(/7/)).toBeInTheDocument();
  expect(view.baseElement).toMatchInlineSnapshot(`
<body>
  <div>
    <div
      class="flex bg-white p-4 rounded-2xl mb-4 shadow-md"
    >
      <a
        href="/wilders/1"
      >
        <img
          alt="Dave"
          class="h-16 w-16 rounded-full mr-6"
          src="avatar.png"
        />
      </a>
      <div
        class="flex justify-between w-full  min-w-[200px]"
      >
        <div
          class="flex flex-col"
        >
          <a
            href="/wilders/1"
          >
            <h3
              class="font-semibold"
            >
              Dave
            </h3>
          </a>
          <ul
            class="flex flex-wrap"
          >
            <li
              class="transition-all duration-500 flex items-center mr-4 mb-3 bg-black/5 hover:bg-black/10 p-1 pl-2 pr-2 rounded-xl cursor-pointer border hover:border-black/5 "
            >
              JS
              <div
                class="items-center flex bg-black/30 text-white rounded-[9999px] h-[25px] pr-2 pl-2 justify-center ml-2 transition-all duration-1000"
              >
                <div
                  class="transition-all duration-1000"
                >
                  7
                </div>
              </div>
            </li>
            <li
              class="transition-all duration-500 flex items-center mr-4 mb-3 bg-black/5 hover:bg-black/10 p-1 pl-2 pr-2 rounded-xl cursor-pointer border hover:border-black/5 "
            >
              PHP
              <div
                class="items-center flex bg-black/30 text-white rounded-[9999px] h-[25px] pr-2 pl-2 justify-center ml-2 transition-all duration-1000"
              >
                <div
                  class="transition-all duration-1000"
                >
                  3
                </div>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="flex flex-col min-w-[40px]"
        >
          <a
            href="/wilders/1/edit"
          >
            <button
              class="mb-2 w-full"
            >
              ✏️
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`);
});
