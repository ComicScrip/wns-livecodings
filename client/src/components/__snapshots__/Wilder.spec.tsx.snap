// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wilder component should properly render the wilder passed in prop 1`] = `
<body>
  <div>
    <div
      class="flex bg-white p-4 rounded-2xl mb-4 shadow-md"
    >
      <a
        href="/wilders/1"
      >
        <img
          alt="Dave"
          class="h-16 w-16 rounded-full mr-6"
          src="avatar.png"
        />
      </a>
      <div
        class="flex justify-between w-full  min-w-[200px]"
      >
        <div
          class="flex flex-col"
        >
          <a
            href="/wilders/1"
          >
            <h3
              class="font-semibold"
            >
              Dave
            </h3>
          </a>
          <ul
            class="flex flex-wrap"
          >
            <li
              class="transition-all duration-500 flex items-center mr-4 mb-3 bg-black/5 hover:bg-black/10 p-1 pl-2 pr-2 rounded-xl cursor-pointer border hover:border-black/5 "
            >
              JS
              <div
                class="items-center flex bg-black/30 text-white rounded-[9999px] h-[25px] pr-2 pl-2 justify-center ml-2 transition-all duration-1000"
              >
                <div
                  class="transition-all duration-1000"
                >
                  7
                </div>
              </div>
            </li>
            <li
              class="transition-all duration-500 flex items-center mr-4 mb-3 bg-black/5 hover:bg-black/10 p-1 pl-2 pr-2 rounded-xl cursor-pointer border hover:border-black/5 "
            >
              PHP
              <div
                class="items-center flex bg-black/30 text-white rounded-[9999px] h-[25px] pr-2 pl-2 justify-center ml-2 transition-all duration-1000"
              >
                <div
                  class="transition-all duration-1000"
                >
                  3
                </div>
              </div>
            </li>
          </ul>
        </div>
        <div
          class="flex flex-col min-w-[40px]"
        >
          <a
            href="/wilders/1/edit"
          >
            <button
              class="mb-2 w-full"
            >
              ✏️
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
